UIController.cs

```diff
@@ -0,0 +1,103 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.UI;
+
+public class UIController : MonoBehaviour
+{
+    private bool ExposedMyCapi = false;
+    public GameObject DragAndMeasureTool;
+    public GameObject AutoPlotterGraph;
+    public GameObject SliderMatch;
+    public GameObject GraphView;
+    public GameObject KeplerGraphView;
+    public GameObject TopDownSlider;
+    public GameObject ButtonContainer;
+    public GameObject ClickAndDragButtons;
+    public GameObject FadePanel;
+    public Slider MyDistanceSlider;
+    public Slider MyTimeSlider;
+    public string AutoPlotterGraphAnim;
+    public string TopDownViewAnim;
+    public string GraphViewAnim;
+    public string KeplerGraphViewAnim;
+    public string FadePanelAnim;
+    public string DragAndMeasureAnim;
+
+    // Use this for initialization
+    void Start ()
+    {
+		
+	}
+	
+	// Update is called once per frame
+	void Update ()
+    {
+		
+	}
+
+    public void ExposeMyCapi()
+    {
+        if (!ExposedMyCapi)

```

Comment:
the only thing that feels awkward about this is that it has so many dependencies. This is why instead of having one class that needs to have lots of dependencies to expose capi, you have one class that stores your state, and make it the dependency of many things, including one class that exposes the relevant data in the data store to capi